tests/functional/heketi/test_heketi_device_operations.py

```diff
@@ -696,3 +710,86 @@ def test_device_settags_tier_option(self):
+
+        try:
+            gluster_server_0 = list(g.config["gluster_servers"].values())[0]
+            manage_hostname = gluster_server_0["manage"]
+            add_device_name = gluster_server_0["additional_devices"][0]
+        except (KeyError, IndexError):
+            self.skipTest(
+                "Additional disk is not specified for node with following "
+                "hostnames and IP addresses: {}, {}".format(
+                    gluster_server_0.get('manage', '?'),
+                    gluster_server_0.get('storage', '?')))
+
+        # Get nodes info
+        node_id_list = heketi_node_list(h_client, h_server)
+        node_info_list = []
+        for node_id in node_id_list[:3]:
+            node_info = heketi_node_info(
+                h_client, h_server, node_id, json=True)
+            node_info_list.append(node_info)
+
+        # Disable 4th and other nodes
+        if len(node_id_list) > 3:
+            for node_id in node_id_list[3:]:
+                heketi_node_disable(h_client, h_server, node_id)
+                self.addCleanup(
+                    heketi_node_enable, h_client, h_server, node_id)
+
+        # Create volume when 3 nodes are online
+        vol_size, vol_count = 2, 4
+        for _ in range(vol_count):
+            vol_info = heketi_blockvolume_create(
+                h_client, h_server, vol_size, json=True)
+            self.addCleanup(
+                heketi_blockvolume_delete, h_client, h_server, vol_info['id'])
+
+        # Get node ID of the Gluster hostname
+        topology_info = heketi_topology_info(
+            h_client, h_server, json=True)
+        self.assertIsNotNone(topology_info, "Failed to get topology info")
+        self.assertIn(
+            "clusters", topology_info.keys(), "Failed to get cluster "
+            "details from topology info")
+        node_list = topology_info["clusters"][0]["nodes"]
+        self.assertTrue(
+            node_list, "Cluster info command returned empty list of nodes")
+
+        node_id = None
+        for node in node_list:
+            if manage_hostname == node['hostnames']["manage"][0]:
+                node_id = node["id"]
+                break
+        self.assertTrue(
+            node_id, "No information about node id for {}".format(
+                manage_hostname))
+
+        # Add extra device, then remember it's ID and size
+        device_id_new, _ = self._add_heketi_device(
+            add_device_name, node_id)
+
+        # Remove one of the existing devices on node except new device
+        device_details = []
+        node_info_after_addition = heketi_node_info(
+            h_client, h_server, node_id, json=True)
+        for device in node_info_after_addition["devices"]:
+            if device["name"] != add_device_name:
+                heketi_device_disable(
+                    h_client, h_server, device["id"])
+                device_details.append((device["name"], device["id"]))
```

Comment:
I would suggest using a dictionary here for storing device details

Reply:
Not required, as it's only two keys and used at only one place.

Reply:
So here then the list is also not required, use 2 variable